### PR TITLE
Resolve keras_tensor case while loop + enable circom of bool-related functions

### PR DESCRIPTION
**Pull request by JernKunpittaya**
- Fix keras_tensor that results in single number but still preserve the dimension e.g. TFAdd(1), it should be assigned in for-loop as if constant when used as input, so we will enforce its index to be 0. 
- Make compile_and_check support general multiple-tensors input for test
- Now support boolean-related operations as follows

    Equal, Lessthan, Greaterthan (including <=, >= as well)
    torch.logical_not
    torch.logical_or
    torch.logical_and
    All these above shown in n test_comp.py

    torch.where (see note below)
- Modify Mpc.circom & layer to contain config accordingly

Note:
- Now, no enforcement that the inputs for AND, OR, NOT must be boolean yet. Since we cannot define === in circom-2-arith 
- TFCast in mpc.circom —> do nothing yet for now —> since now we just do all in integer, and no concept of boolean in circom yet. Can change once we introduce floating point and their scale
- Where operations is almost done. However, please take a look at test_where.py where most tests pass except one that somehow input data swap places. I suspect it may come from MPSPDZ parser since the circom and onnx file are correct. @mhchia Could you help investigate? Thanks :))

### CODE REVIEW

File: zkstats/onnx2circom/mpc.circom [hunk 1]
```diff
@@ -37,6 +37,78 @@ template TFDiv(nElements) {
...
+    for (var i = 0; i<nElements; i++){
+        out[i] <== left[i] < right[i];
+    }
+}
+
+template TFNot(nElements) {
+    signal input in[nElements];
+    signal output out[nElements];
+    for (var i = 0; i<nElements; i++){
+        out[i] <== 1-in[i];
```
Comment (mhchia) [inline zkstats/onnx2circom/mpc.circom:71]: I think we can add a `TODO` label so we can revisit
```suggestion
        # TODO: should have a check for `in` somewhere in circom or MP-SPDZ 
        out[i] <== 1-in[i];
```
